posit-dev/positron
# Bundle Quarto with Positron

This change bundles the Quarto CLI and all of its requisite tooling (deno, typst, etc.) with Positron, replacing the former Pandoc-only approach. Quarto is currently version-pinned at 1.5.55. 

The Quarto bundle is stored in the `quarto` subdirectory of the app root. 

Addresses #3705.

> [!IMPORTANT]
> This change alone is not sufficient to allow the Quarto extension to discover the copy of Quarto bundled in Positron; the Quarto extension's search path is not configurable, so a separate change will need to be made to the Quarto extension to include the Positron app directory in the search path. I'll make that change once this one has landed successfully.

### QA Notes

- This change needs to be tested on all 3 platforms.
- The Quarto extension _does_ look on the `PATH` for Quarto, so if you're feeling brave you can cause it to use this bundled version of Quarto by putting it on your `PATH`.
- This change uses a different version of Pandoc and puts Pandoc in a new location. Make sure that `RSTUDIO_PANDOC` points to a folder that contains the pandoc executable, and that `rmarkdown::find_pandoc()` and the "Render R Markdown" command are able to use the new Pandoc.
- Both the Quarto extension and R Markdown package will find Quarto/Pandoc on the `PATH`, so to ensure you're testing against the embedded copy of Quarto and Pandoc in Positron, make sure to remove any other installations of Quarto and Pandoc on your `PATH` (or temporarily modify your `PATH` so they aren't discovered)
- It is expected that this change will dramatically increase the size of the installers.
